cmd/troubleshoot/cli/run.go

```diff
@@ -258,6 +267,47 @@ func loadSpec(v *viper.Viper, arg string) ([]byte, error) {
 		return nil, fmt.Errorf("%s is not a URL and was not found (err %s)", arg, err)
 	}
 
+	spec, err := loadSpecFromURL(v, arg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get spec from URL")
+	}
+	return spec, nil
+}
+
+func loadSpecFromSecret(v *viper.Viper, arg string) ([]byte, error) {
+	// format secret/namespace-name/secret-name
+	pathParts := strings.Split(arg, "/")
+
+	if len(pathParts) != 3 {
```

Comment:
can this be in the caller, so that if it hits this code only if there are 3 parts?